jdaviz/configs/default/plugins/subset_plugin/subset_plugin.vue

```diff
@@ -22,23 +22,39 @@
       </v-col>
     </v-row>
 
+    <!-- Composite region cannot be edited, so just grab first element. -->
+    <div v-if="is_editable">
+      <v-row v-for="(item, index2) in subset_definitions[0]" no-gutters>
+        <v-col>
+          <v-text-field
+            :label="item.name"
+            v-model.number="item.value"
+            type="number"
+            :disabled="item.name=='Angle'"
+          ></v-text-field>
```

Comment:
I **think** it would be something like this:

```suggestion
          <v-text-field
            :label="item.name"
            v-model.number="item.value"
            @keyup.enter="update_subset"
            type="number"
            :disabled="item.name=='Angle'"
          ></v-text-field>
```

Injecting it in on myside seems to do the trick

Reply:
Hmm... but this will:

(a) make this plugin behavior different from the other ones; and
(b) might prematurely update a shape when someone accidentally hits enter but really want to also update other fields?

Reply:
> (a) make this plugin behavior different from the other ones

I'm not sure if there's a consistent pattern here. Take, for instance, the `Line Width` field for plot options: when I update the text field, it _immediately_ updates the line width, doesn't even wait for the user. I'd argue that it might be annoying to the user to have the field update while the user is trying to type the bounds, hence why I proposed `Enter` to accept. Though now that I say it, is the `UPDATE` button even necessary in this context? Why not just have it accept automatically? Which of course leads to...

> want to also update other fields?

Is there a problem with updating one field at a time? I don't necessarily see it as an issue if I update the `xcenter` and THEN the `ycenter` rather than changing both fields simultaneously

Reply:
Come to think about it, I wonder if the disconnect here has to do with perspective. For `Line Width` I think of that field as an actual representation of the value, hence when I change the number, I expect it to immediately update. For `Model Fit`, it's more like a command, so I fill out all the necessary boxes and then tell it to activate via the button click.

As I tested it now, it appears to be more of the "command" like paradigm, where the user fills out all the information and then actively requests the plugin to "UPDATE" the subset by pressing the button. I suppose my request is really wondering whether it would be better to change this to the "representation" paradigm to have it behave more like plot options, like `Line Width` for instance.

Reply:
Subset update may be expensive, because it might affect other calculations etc. You would only want to update it when you are sure you want to update it.

Reply:
> Subset update may be expensive

I'm trying to think of cascading events that would happen on a SubsetUpdate message. Most of the heavy ones are still locked behind a button press (like Aperture Photometry and Model Fitting). The only ones I can think of are the Cubeviz spectrum collapse and line analysis, and the latter is only if the plugin is open. I feel like "Enter to Accept" is a QoL feature I'd be happy to accept in exchange. Are there other cases I'm not appreciating?

Reply:
I feel like this needs broader discussion and is out of scope here. If we go with this, it will apply to other plugins too anyway.

Reply:
I definitely think we need some standards defined here as we already use a few patterns throughout the plugins.  But in general, I'd agree that they _should_ all fit into either an action button or live-updating (but never both)... and the decision between which to use are the points brought up by both of you: (a) whether live-feedback between changing fields is useful and (b) whether the action is too expensive to update in real-time.  The conflict we currently have is that some fields that are live-updating update on keystroke and others when the input element loses focus - but that is definitely out of scope for this PR (but I'd argue deciding whether this PR should be live-updating when losing focus might not be).

Reply:
> deciding whether this PR should be live-updating when losing focus might not be (out of scope)

You explained it MUCH more eloquently than I; thanks Kyle!

Reply:
here are the options for updating a traitlet from a `v-text-field` with `type="number"` that are used elsewhere already:
* live updating on keypress: use `v-model`  (see slice plugin for a good example) and then `@observe` in python.  We use this quite often and it mimics what glue widgets do, so unless there are performance limitations, this might be worth trying first.
* update on change (loss of focus) or mousepress on the up/down buttons: use `@change` and `@mousedown` to call a JS or python method to update the traitlet manually (play_pause widget does something similar - but this also uses `v-model` and just triggers acting on it at those times, you could do the same to always have the traitlet update but only trigger the “update” when this happens, or if you want to have an `@observe` still in python, then you might need extra traitlets).

Reply:
Re: lag -- I tried with 6 4k x 2k and Imviz is already super laggy (the lag from Subset update doesn't matter much at that point because the whole thing is unusable, which is out of scope here). With 4 of those images, Imviz lag is tolerable but you can really see the lag in Subset shape catching up with each click to Update. So, I am still not convinced live-updating is a good idea.

You can try for yourselves like this:

1. Open up Imviz example notebook.
2. Load the example images twice (so you get 4 total) with this additional call:

```python
imviz.load_data(acs_47tuc_1, data_label='acs_47tuc_3')
imviz.load_data(acs_47tuc_2, data_label='acs_47tuc_4')
```

3. Draw a Subset.
4. Use this new edit feature and change some field of the Subset, click Update. See lag.

Reply:
p.s. This step is also important: `imviz.link_data(link_type='wcs', wcs_use_affine=False)` (links them by WCS without fast approximation)

Reply:
Down the road I think we can either look into improving the performance or perhaps have this plugin check the number of links (and if fast approximation is enabled) and switch itself between live-updating and the update button.  But I'm ok considering that as a follow-up effort and moving forward with the PR with the update button.

Reply:
Having the button come and go is even more confusing than just allowing live-update unconditionally. But I agree this can be future work.

Please approve if the PR is okay as-is or if there are more changes needed here. Thanks!